deploy/helm/kubecf/templates/_config.tpl

```diff
@@ -0,0 +1,263 @@
+|
+| Lookup $path under $context and return either the value found, or the empty string.
+|
+| $context is either is an object (or array). It normally starts out as $.kubecf.config
+| or $.kubecf.manifest, but moves down the tree as _lookup calls itself recursively.
+|
+| $path is a list of properties to look up successively, e.g. ("stacks" $stack "releases").
+|
+| When the found value is `nil`, we still return the empty string and not a stringified
+| nil (which would be "<no data>"), because helm prints it as an empty string, but
+| otherwise treats as a non-empty string for all other purposes.
+|
+| $kubecf.retval is also set to the found entry in case the caller needs an object and
+| not just a string. This can also be used to disambiguate between nil and the empty
+| string (which can mean: not found).
+|
+| When $context is an array, then _lookup will look for an array element that has a "name"
+| property matching the first element of $path.
+|
+| Example: Looking for "instance_groups.api.jobs.cloud_controller_ng" in $.kubecf.manifest
+| is equivalent to the "/instance_groups/name=api/jobs/name=cloud_controller_ng" path in
+| JSON-patch.
++-----------------------------------------------------------------------------------------
+| Not implemented:
+|
+| When the full $path cannot be found, and the $.kubecf.required variable is true, then
+| _lookup should throw an error instead of returning an empty string. It could use
+| $.kubecf.query to provide a helpful error message. With the introduction of a
+| validation schema, this may no longer be required.
+|
+| There is no mechanism to search array $context for any other properties but "name".
+| It can easily be done on the caller side as well, but otherwise should be easy to
+| add here as well. So far we don't need it.
+==========================================================================================
+*/}}
+{{- define "_config._lookup" }}
+  {{- $kubecf := index . 0 }}
+  {{- $context := index . 1 }}
+  {{- $path := slice . 2 }}
+  {{- $key := (first $path) }}
+  {{- if kindIs "slice" $context }}
+    {{- $_ := set $kubecf "retval" nil }}
+    {{- range $context }}
+      {{- if eq (index . "name") $key }}
+        {{- $_ := set $kubecf "retval" . }}
+      {{- end }}
+    {{- end }}
+  {{- else }}
+    {{- $_ := set $kubecf "retval" (index $context $key) }}
+  {{- end }}
+  {{- if $kubecf.retval }}
+    {{- if gt (len $path) 1 }}
+      {{- include "_config._lookup" (concat (list $kubecf $kubecf.retval) (rest $path)) }}
+    {{- else }}
+      {{- $kubecf.retval }}
+    {{- end }}
+  {{- end }}
+{{- end }}
+
+
+{{- /*
+==========================================================================================
+| _config.condition (list $ $condition)
++-----------------------------------------------------------------------------------------
+| Evaluates $condition and returns either the string "true" or the empty string.
```

Comment:
The description here may be misleading depending on how `the string "true"` part is read. IIUC, when the condition is rendered, it is up to the caller to quote the boolean or not. I.e. the rendered value is not a string per se.

Reply:
I was going to argue that it doesn't matter what exact value the template renders to return `true` as long as it evaluates to true in a boolean context.

But now I have an idea how to implement sub-expression grouping with parenthesis, and that may require this template to render the strings `true` and `false`.

Reply:
Fixed (I think) in e895e02. Let me know if it is still ambiguous.